src/main/java/org/folio/rest/impl/base/AbstractEmail.java

```diff
@@ -202,26 +199,54 @@ protected Future<Configurations> lookupConfiguration(Map<String, String> request
           return succeededFuture(config);
         }
         String errorMessage = String.format(ERROR_LOOKING_UP_MOD_CONFIG,
-          url, response.statusCode(), response.bodyAsString());
+          path, response.statusCode(), response.bodyAsString());
         logger.error(errorMessage);
         return failedFuture(new ConfigurationException(errorMessage));
       });
   }
 
-  protected Future<Configurations> validateConfiguration(Configurations configurations) {
-    if (isIncorrectSmtpServerConfig(configurations)) {
-      String errorMessage = String.format(ERROR_MIN_REQUIREMENT_MOD_CONFIG,
-        REQUIREMENTS_CONFIG_SET);
-      logger.error(errorMessage);
-      return failedFuture(new SmtpConfigurationException(errorMessage));
-    }
+  private Future<SmtpConfiguration> copyConfigurationAndDeleteFromModConfig(
+    Configurations configurations, Map<String, String> requestHeaders) {
+
+    return succeededFuture(configurations)
+      .map(EmailUtils::convertSmtpConfiguration)
+      .compose(EmailUtils::validateSmtpConfiguration)
+      .compose(smtpConfigurationService::createSmtpConfiguration)
+      .compose(smtpConfig -> deleteEntriesFromModConfig(smtpConfig, configurations, requestHeaders));
+  }
 
-    return succeededFuture(configurations);
+  private Future<SmtpConfiguration> deleteEntriesFromModConfig(
+    SmtpConfiguration validSmtpConfiguration, Configurations configurationsToDelete,
+    Map<String, String> requestHeaders) {
+
+    logger.warn("Removing configuration from mod-config");
+
+    OkapiClient okapiClient = new OkapiClient(vertx, requestHeaders, webClientOptions);
```

Comment:
We instantiate OkapiClient twice. Can we do it in constructor and store it a field?

Reply:
Short answer is no, we don't create an instance of this class for every request. Instead, processEmail() method is called with okapi headers